android/filament-android/src/main/cpp/MaterialInstance.cpp

```diff
@@ -260,6 +441,48 @@ Java_com_google_android_filament_MaterialInstance_nSetParameterTexture(
     env->ReleaseStringUTFChars(name_, name);
 }
 
+extern "C"
+JNIEXPORT jobject JNICALL
+Java_com_google_android_filament_MaterialInstance_nGetParameterTexture(
+        JNIEnv *env, jclass, jlong nativeMaterialInstance, jstring name_) {
+    MaterialInstance* instance = (MaterialInstance*) nativeMaterialInstance;
+    Texture const* texture;
+    TextureSampler sampler;
+
+    const char *name = env->GetStringUTFChars(name_, 0);
+    auto success = instance->getParameter(name, texture, sampler);
+    env->ReleaseStringUTFChars(name_, name);
+
+    if (!success) {
+        return NULL;
+    }
+
+    const auto jTextureClass = env->FindClass("com/google/android/filament/Texture");
```

Comment:
These need to be looked up once at initialization, not every time the function is called